packages/drafts-realm/opportunity-form.gts

```diff
@@ -0,0 +1,602 @@
+  ] as Array<CategorySignature>;
+
+  @action updateCategory(type: { name: string }) {
+    this.selectedCategory = type;
+    this.args.model.category = type.name;
+  }
+
+  <template>
+    <CardContainer @displayBoundaries={{false}} class='card-container'>
+
+      <BoxelSelect
+        @searchEnabled={{true}}
+        @searchField='name'
+        @placeholder={{this.categoryPlaceholder}}
+        @selected={{this.selectedCategory}}
+        @onChange={{this.updateCategory}}
+        @options={{this.categoryOptions}}
+        class='select'
+        as |item|
+      >
+        <div>{{item.name}}</div>
+      </BoxelSelect>
+
+    </CardContainer>
+
+    <style>
+      .select {
+        padding: var(--boxel-sp-xs);
+        background-color: white;
+      }
+    </style>
+  </template>
+}
+
+class StageField extends FieldDef {
+  @field category = contains(StringField, {
+    description: `Selected Category`,
+  });
+
+  static edit = EditSecForStageField;
+  static embedded = EmbeddedSecForStageField;
+}
+
+/* Account */
+class EditSecForAccountField extends Component<typeof AccountField> {
+  @tracked selectedAccount = {
+    name: this.args.model.name || 'Select',
+  };
+
+  @tracked accountPlaceholder = this.args.model.name || 'Select';
+
+  get getAccountsNames() {
+    let allAccounts = this.args.model.accounts || [];
+    return allAccounts.map((o) => ({ name: o.accountName }));
+  }
+
+  @action updateAccount(type: { name: string }) {
+    this.selectedAccount = type;
+    this.args.model.name = type.name;
+  }
+
+  <template>
+    <CardContainer @displayBoundaries={{false}} class='card-container'>
+
+      <BoxelSelect
+        @placeholder={{this.accountPlaceholder}}
+        @selected={{this.selectedAccount}}
+        @onChange={{this.updateAccount}}
+        @options={{this.getAccountsNames}}
+        class='select'
+        as |item|
+      >
+        <div>{{item.name}}</div>
+      </BoxelSelect>
+
+    </CardContainer>
+
+    <style>
+      .select {
+        padding: var(--boxel-sp-xs);
+        background-color: white;
+      }
+    </style>
+  </template>
+}
+
+class EmbeddedSecForAccountField extends Component<typeof AccountField> {
+  <template>
+    <CardContainer @displayBoundaries={{false}} class='card-container'>
+      {{this.args.model.name}}
+    </CardContainer>
+
+    <style>
+      .card-container {
+        background: transparent;
+      }
+    </style>
+  </template>
+}
+
+class AccountField extends FieldDef {
+  @field accounts = linksToMany(() => CrmAccount, {
+    description: `CRM Accounts`,
+  });
+  @field name = contains(StringField, {
+    description: `Account Name`,
+  });
+
+  static edit = EditSecForAccountField;
+  static embedded = EmbeddedSecForAccountField;
+}
+
+/* Amount */
+class EditSecForAmountField extends Component<typeof AmountField> {
+  @tracked currencyOptions = ['USD', 'RM'];
+  @tracked currencyPlaceHolder = 'RM';
+  @tracked selectedCurrency = this.args.model.currency || 'RM';
```

Comment:
TODO: These 3 tracked properties are very related. Justin to discuss and build another API.